extensions/standard-processors/tests/unit/ListenTcpTests.cpp

```diff
@@ -134,60 +147,64 @@ TEST_CASE("Test ListenTCP with SSL connection", "[ListenTCP][NetworkListenerProc
 
   SECTION("Without client certificate verification") {
     SECTION("Client certificate not required, Client Auth set to NONE by default") {
+      ssl_context_service->enable();
+      port = scheduleProcessorOnRandomPort(controller, listen_tcp);
       SECTION("sending through IPv4", "[IPv4]") {
-        endpoint = asio::ip::tcp::endpoint(asio::ip::address_v4::loopback(), PORT);
+        endpoint = asio::ip::tcp::endpoint(asio::ip::address_v4::loopback(), port);
       }
       SECTION("sending through IPv6", "[IPv6]") {
         if (utils::isIPv6Disabled())
           return;
-        endpoint = asio::ip::tcp::endpoint(asio::ip::address_v6::loopback(), PORT);
+        endpoint = asio::ip::tcp::endpoint(asio::ip::address_v6::loopback(), port);
       }
     }
     SECTION("Client certificate not required, but validated if provided") {
       REQUIRE(controller.plan->setProperty(listen_tcp, ListenTCP::ClientAuth.getName(), "WANT"));
+      ssl_context_service->enable();
+      port = scheduleProcessorOnRandomPort(controller, listen_tcp);
       SECTION("sending through IPv4", "[IPv4]") {
-        endpoint = asio::ip::tcp::endpoint(asio::ip::address_v4::loopback(), PORT);
+        endpoint = asio::ip::tcp::endpoint(asio::ip::address_v4::loopback(), port);
       }
       SECTION("sending through IPv6", "[IPv6]") {
         if (utils::isIPv6Disabled())
           return;
-        endpoint = asio::ip::tcp::endpoint(asio::ip::address_v6::loopback(), PORT);
+        endpoint = asio::ip::tcp::endpoint(asio::ip::address_v6::loopback(), port);
       }
     }
-    ssl_context_service->enable();
-    controller.plan->scheduleProcessor(listen_tcp);
 
     expected_successful_messages = {"test_message_1", "another_message"};
     for (const auto& message: expected_successful_messages) {
-      REQUIRE(utils::sendMessagesViaSSL({message}, endpoint, minifi::utils::file::concat_path(executable_dir, "resources/ca_A.crt")));
+      check_no_error(utils::sendMessagesViaSSL({message}, endpoint, minifi::utils::file::concat_path(executable_dir, "resources/ca_A.crt")));
```

Comment:
should we rename this to `CHECK_NO_ERROR`, to mimic "checker" macros?

Reply:
sure thing, I've also moved it to test utils.h in https://github.com/apache/nifi-minifi-cpp/pull/1457/commits/70d6a798090869c4c01b2dd7dc84eeaf9dcef2a3#diff-eb32155f3c36cff6b7fb634b81ae2288627fae11502496cee252b9493e45bb4aR221-R223

Reply:
I'd prefer to keep it lowercase, since it's not a macro.

Reply:
I've sidestepped the issue by creating a MatchesSuccess Matcher and use that. https://github.com/apache/nifi-minifi-cpp/commit/fb2cba6cb1b8eab262e1d17b5d67d69ed0e1e9a3